src/components/test-element-settings/TestElementSettings.test.tsx

```diff
@@ -0,0 +1,65 @@
+import * as React from "react";
+import TestElementSettings from "./TestElementSettings";
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { ITestElementSettings } from "@/AppSettings";
+import ElementManager from "@/ElementManager";
+
+const TEST_SETTING: ITestElementSettings = {
+  atomic: 26,
+  enabled: true,
+  stats: {
+    right: 0,
+    times: 0,
+    wrong: 0,
+  },
+};
+
+describe("TestElementSettings", () => {
+  beforeAll(() => {
+    ElementManager.loadElements();
+  });
+
+  it("shows the elements symbol", () => {
+    render(<TestElementSettings setting={TEST_SETTING} />);
+
+    expect(screen.getByText("Fe")).toBeInTheDocument();
+  });
+
+  it("shows the elements atomic number and name", () => {
+    render(<TestElementSettings setting={TEST_SETTING} />);
+
+    expect(screen.getByText("26. Iron")).toBeInTheDocument();
+  });
+
+  it("shows the elements group", () => {
+    render(<TestElementSettings setting={TEST_SETTING} />);
+
+    expect(screen.getByText("Transition metals")).toBeInTheDocument();
+  });
+
+  it("calls onClick when the element is clicked", () => {
+    const onClick = jest.fn();
```

Comment:
why is this mocked?

Reply:
Because otherwise, I have no way to test that the `onClick` prop is called when the element is clicked, how would you test that the `onClick` prop is called when it should?

Reply:
see https://github.com/OpenSourceRaidGuild/atom-pwa/pull/52#discussion_r584223662